cdap-master/src/main/java/co/cask/cdap/master/startup/MasterStartupTool.java

```diff
@@ -163,7 +165,13 @@ static Injector createInjector(CConfiguration cConf, Configuration hConf) {
         @Override
         protected void configure() {
           bind(FileContext.class).toProvider(FileContextProvider.class).in(Scopes.SINGLETON);
-          bind(LocationFactory.class).toProvider(RootLocationFactoryProvider.class).in(Scopes.SINGLETON);
+        }
+
+        @Provides
+        @Singleton
+        private LocationFactory providesLocationFactory(Configuration hConf, CConfiguration cConf, FileContext fc) {
```

Comment:
This addition is temporary. Should be just using `LocationRuntimeModule`.getDistributedModule`. However it requires more refactoring of the location module (part II of the refactoring).